### PR TITLE
Add information about storing vanilla functions in state object

### PR DESCRIPTION
**Pull request by thelinuxlich**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -168,6 +168,21 @@ function Foo() {
   const [snapshot, state] = useLocalProxy({ count: 0, text: 'hello' })
 ```
 
+### Adding vanilla functions to the state object
+
+For organization' sake, you might want to add functions in your state object. Just be aware that these functions won't have any reactivity, as vanilla Javascript, they work through explicit invocation.
+
```
Comment (AjaxSolutions) [inline readme.md:174]: I like this [example](https://twitter.com/0xca0a/status/1329023210768265216) better.

Comment (thelinuxlich) [reply readme.md:174]: I don't see what you mean, in that example the user is assigning to a reactive field the result of a fetch, a Promise.

Comment (AjaxSolutions) [reply readme.md:174]: OK I see your point, but I prefer the _addCity_ example in this [video](https://youtu.be/XnlECbwcJZ0?t=461).  I'd rather document Jack Herrington's approach, but both can be documented if you feel strongly about it.

Comment (thelinuxlich) [reply readme.md:174]: I made a small change, but I feel the example should be kept simple and concise.

**Review by dai-shi — COMMENTED** [review]
I have two comments on this.
- we'd like to keep readme simple and concise as you said, and this is slightly different from  others. this is more like best practice or one of it, using the library. I agree it should be documented somehow, but it we were to add this here, there would be so many others to add.
- I'm still considering which to recommend a) using `state` in `fetch` and allowing both `snapshot.fetch()` and `state.fetch()`, or b) using `this` in `fetch` and only allowing `state.fetch()`.

Meanwhile, I would be more comfortable to create a wiki page and link there from readme.

**Comment by thelinuxlich** [comment]
Why use this in fetch when state is accessible?

**Comment by thelinuxlich** [comment]
Instead of this we could have a section explaining what types of variables aren't wrapped with proxies(Function, Set), probably within the _ref_ documentation

**Comment by dai-shi** [comment]
> Why use this in fetch when state is accessible?

Because it looks more self organized? You could even create the object before `proxy()`.

**Comment by dai-shi** [comment]
> Instead of this we could have a section explaining what types of variables aren't wrapped with proxies(Function, Set, Map, Date), probably within the _ref_ documentation

That sounds good. I still like to use `<overview>` as it gets long.

How do you like the idea of wiki pages for this (this issue) kind of best practices?

**Comment by thelinuxlich** [comment]
I never used the overview tag, do you mean summary and details? IMO makes some important info hidden, like React-Easy-State docs https://github.com/RisingStack/react-easy-state

**Comment by dai-shi** [comment]
It would depend. It may hide other basic info less visible, otherwise.

yes, details/summary. why did I think overview???

**Comment by thelinuxlich** [comment]
I think the docs need just a ToC. I will close this PR.

**Pull request closed without merging**